cmake/podioBuild.cmake

```diff
@@ -27,6 +27,10 @@ macro(podio_set_rpath)
     if("${isSystemDir}" STREQUAL "-1")
       set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${LIBDIR}")
     endif("${isSystemDir}" STREQUAL "-1")
+    # Make sure to actualy set RPATH and not RUNPATH by disabling the new dtags
+    # explicitly. Set executable and shared library linker flags for this
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--disable-new-dtags")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--disable-new-dtags")
```

Comment:
And I hope this "fixes" the Ubuntu CI builds. The problem is that on Ubuntu setting `RPATH`, actually sets `RUNPATH` which has lower precedence than `LD_LIBRARY_PATH`. Debian based systems, seem to have different default linker flags, which set `RPATH` instead of `RUNPATH`.

Following a few stackoverflow threads, there seems to be no real consensus which one is the "correct" one. In any case these changes **force `RPATH`, i.e. there is now way to override via the environment**. At least for Key4hep it should not really be a problem, since we always only have one podio in the environment.